adapters/repos/db/vector/hnsw/commit_logger_snapshot.go

```diff
@@ -0,0 +1,1079 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package hnsw
+
+import (
+	"bufio"
+	"encoding/binary"
+	"fmt"
+	"hash"
+	"hash/crc32"
+	"io"
+	"math"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/weaviate/weaviate/adapters/repos/db/vector/compressionhelpers"
+	entcfg "github.com/weaviate/weaviate/entities/config"
+	"github.com/weaviate/weaviate/entities/diskio"
+	enterrors "github.com/weaviate/weaviate/entities/errors"
+)
+
+const checkpointChunkSize = 100_000
+
+const (
+	SnapshotCompressionTypePQ = iota + 1
+	SnapshotCompressionTypeSQ
+)
+
+// feature flag to disable snapshots
+func snapshotsDisabled() bool {
+	return entcfg.Enabled(os.Getenv("DISABLE_SNAPSHOTS"))
```

Comment:
```suggestion
	return entcfg.Enabled(os.Getenv("DISABLE_HNSW_SNAPSHOTS"))
```

Reply:
Also again ideally in environment.go to be centrally managed especially when scheduling is involved.